app/src/main/java/com/konkuk/arabyte_aos/presentation/ui/component/ArabyteTopAppBar.kt

```diff
@@ -0,0 +1,96 @@
+package com.konkuk.arabyte_aos.presentation.ui.component
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.unit.dp
+import com.konkuk.arabyte_aos.presentation.type.ArabyteTopAppBarType
+import com.konkuk.arabyte_aos.presentation.util.modifier.noRippleClickable
+import com.konkuk.arabyte_aos.ui.theme.ArabyteTheme
+
+/**
+ * 4가지 타입의 상단 앱 바를 나타내는 컴포넌트입니다.
+ *
+ * @param appBarType 4가지 앱 바의 유형을 정의하는 [ArabyteTopAppBarType]입니다.
+ * @param modifier UI 수정 사항을 추가할 수 있는 Modifier입니다.
+ * @param onBackClick 좌측 back 버튼 클릭 시 호출되는 콜백 함수입니다. (뒤로가기)
+ * @param onOptionalClick 우측 선택적 버튼 클릭 시 호출되는 콜백 함수입니다.
+ *        기본값은 null이며, 제공되지 않을 경우 해당 버튼은 표시되지 않습니다.
+ */
+@Composable
+fun ArabyteTopAppBar(
+    appBarType: ArabyteTopAppBarType,
+    modifier: Modifier = Modifier,
+    onBackClick: () -> Unit,
+    onOptionalClick: (() -> Unit)? = null,
```

Comment:
얘도 민석이 형이 한 것 처럼 이렇게 구현하면 할당할 때 분기처리 안해줘도 되지 않을까요

```suggestion
    onOptionalClick: (() -> Unit) = {},
```